c10/util/Logging.h

```diff
@@ -257,6 +257,32 @@ BINARY_COMP_HELPER(LessEquals, <=)
 #define CAFFE_ENFORCE_GT_WITH_CALLER(x, y, ...) \
   CAFFE_ENFORCE_THAT_IMPL_WITH_CALLER(          \
       Greater((x), (y)), #x " > " #y, __VA_ARGS__)
+
+/**
+ * Very lightweight logging for the first time API usage. It's beneficial for
+ * tracking of individual functionality usage in larger applications.
+ *
+ * In order to ensure light-weightness of logging, we utilize static variable
+ * trick - LogAPIUsage will be invoked only once and further invocations will
+ * just do an atomic check.
```

Comment:
an atomic check on for example a hot-loop, wouldn't that add a few hundred nano-seconds per TensorImpl usage?
Is the logging only enabled if the env variable is set, and the atomic check skippable if it's not?

Reply:
The logging will be enabled all the time in e.g. FB environment. The envvar thing is just for testing in default implementation.

I've tried to measure overhead and if I don't wipe out cache (it's hard to measure with cold cache as the overhead is so tiny), it comes down to something like 0.3ns (nanoseconds) so it's definitely fine. The static var initialization trick is used widely (including for singletons), so it should be cheap to sprinkle around - https://stackoverflow.com/questions/23829389/does-a-function-local-static-variable-automatically-incur-a-branch